### PR TITLE
(Cgt 3262) - Updating CheckYourAnswersConstructor for new PropertyLivedIn flow

### PR DESCRIPTION
**Pull request by euptonqa**

### CODE REVIEW

**Comment by euptonqa** [comment]
I merged in Jane's PropertyLivedIn controller/view/form etc before it was merged into master... I think that's why it's showing up that I've added files that are already in master... (Look at the first commit)

File: app/constructors/YourAnswersConstructor.scala [hunk 1]
```diff
@@ -16,19 +16,27 @@
...
 object YourAnswersConstructor {
 
   def fetchYourAnswers(totalGainAnswersModel: TotalGainAnswersModel,
                        privateResidenceReliefModel: Option[PrivateResidenceReliefModel] = None,
-                       personalAndPreviousDetailsModel: Option[TotalPersonalDetailsCalculationModel] = None): Seq[QuestionAnswerModel[Any]] = {
+                       personalAndPreviousDetailsModel: Option[TotalPersonalDetailsCalculationModel] = None,
+                       propertyLivedInModel: Option[PropertyLivedInModel] = None): Seq[QuestionAnswerModel[Any]] = {
     val salesDetailsRows = SalesDetailsConstructor.salesDetailsRows(totalGainAnswersModel)
     val purchaseDetailsRows = PurchaseDetailsConstructor.getPurchaseDetailsSection(totalGainAnswersModel)
+    val propertyLivedInRow = DeductionDetailsConstructor.propertyLivedInQuestionRow(propertyLivedInModel)
```
Comment (jameshforster) [inline app/constructors/YourAnswersConstructor.scala]: This should be included as part of the deduction details rows, not as a separate row.

File: app/constructors/YourAnswersConstructor.scala [hunk 2]
```diff
@@ -16,19 +16,27 @@
...
     val propertyDetailsRows = PropertyDetailsConstructor.propertyDetailsRows(totalGainAnswersModel)
     val deductionDetailsRows = DeductionDetailsConstructor.deductionDetailsRows(totalGainAnswersModel, privateResidenceReliefModel)
     val personalAndPreviousDetailsRows = PersonalAndPreviousDetailsConstructor.personalAndPreviousDetailsRows(personalAndPreviousDetailsModel)
 
-    salesDetailsRows ++ purchaseDetailsRows ++ propertyDetailsRows ++ deductionDetailsRows ++ personalAndPreviousDetailsRows
+    if(!propertyLivedInRow.isEmpty) {
```
Comment (jameshforster) [inline app/constructors/YourAnswersConstructor.scala]: This logic should sit in the DeductionDetailsConstructor

File: app/controllers/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -140,15 +140,17 @@ trait CheckYourAnswersController extends FrontendController with ValidActiveSess
...
   }
 
+
   val checkYourAnswers: Action[AnyContent] = ValidateSession.async { implicit request =>
 
     for {
       model <- answersConstructor.getNRTotalGainAnswers
       totalGainResult <- calculatorConnector.calculateTotalGain(model)
       prrModel <- getPRRModel(totalGainResult)
+      propertyLivedInModel <-  calculatorConnector.fetchAndGetFormData[PropertyLivedInModel](KeystoreKeys.propertyLivedIn)
```
Comment (jameshforster) [inline app/controllers/CheckYourAnswersController.scala]: This needs to be called before the PRR and used to determine whether the PRR is called in the first place.

File: app/controllers/ReportController.scala [hunk 4]
```diff
@@ -128,13 +128,14 @@ trait ReportController extends FrontendController with ValidActiveSession {
       totalGainResultsModel <- calcConnector.calculateTotalGain(answers)
 
       privateResidentReliefModel <- getPRRModel(hc, totalGainResultsModel.get)
+      propertyLivedInModel <-  calcConnector.fetchAndGetFormData[PropertyLivedInModel](KeystoreKeys.propertyLivedIn)
```
Comment (jameshforster) [inline app/controllers/ReportController.scala]: This needs to be called before the PRR and used to determine whether the PRR is called in the first place.

**Pull request merged**